api/v1alpha1/dnspolicy_types.go

```diff
@@ -0,0 +1,343 @@
+	//
+	// The values accepted are determined by the target dns provider, please refer to the appropriate docs below.
+	//
+	// Route53: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-geo.html
+	// +required
+	DefaultGeo string `json:"defaultGeo,omitempty"`
+}
+
+// DNSPolicyStatus defines the observed state of DNSPolicy
+type DNSPolicyStatus struct {
+	// conditions are any conditions associated with the policy
+	//
+	// If configuring the policy fails, the "Failed" condition will be set with a
+	// reason and message describing the cause of the failure.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// observedGeneration is the most recently observed generation of the
+	// DNSPolicy.  When the DNSPolicy is updated, the controller updates the
+	// corresponding configuration. If an update fails, that failure is
+	// recorded in the status condition
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	HealthCheck *HealthCheckStatus `json:"healthCheck,omitempty"`
+}
+
+var _ common.KuadrantPolicy = &DNSPolicy{}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=direct"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[0].reason`,description="DNSPolicy Status",priority=2
+// +kubebuilder:printcolumn:name="TargetRefKind",type="string",JSONPath=".spec.targetRef.kind",description="Type of the referenced Gateway API resource",priority=2
+// +kubebuilder:printcolumn:name="TargetRefName",type="string",JSONPath=".spec.targetRef.name",description="Name of the referenced Gateway API resource",priority=2
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// DNSPolicy is the Schema for the dnspolicies API
+type DNSPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DNSPolicySpec   `json:"spec,omitempty"`
+	Status DNSPolicyStatus `json:"status,omitempty"`
+}
+
+func (p *DNSPolicy) GetWrappedNamespace() gatewayapiv1.Namespace {
+	return gatewayapiv1.Namespace(p.Namespace)
+}
+
+func (p *DNSPolicy) GetRulesHostnames() []string {
+	return make([]string, 0)
+}
+
+func (p *DNSPolicy) GetTargetRef() gatewayapiv1alpha2.PolicyTargetReference {
+	return p.Spec.TargetRef
+}
+
+func (p *DNSPolicy) Kind() string { return p.TypeMeta.Kind }
+
+// Validate ensures the resource is valid. Compatible with the validating interface
+// used by webhooks
+func (p *DNSPolicy) Validate() error {
+	if p.Spec.TargetRef.Group != gatewayapiv1.GroupName {
+		return fmt.Errorf("invalid targetRef.Group %s. The only supported group is gateway.networking.k8s.io", p.Spec.TargetRef.Group)
+	}
+
+	if p.Spec.TargetRef.Kind != ("Gateway") {
+		return fmt.Errorf("invalid targetRef.Kind %s. The only supported kind is Gateway", p.Spec.TargetRef.Kind)
+	}
```

Comment:
AuthPolicy and RateLimitPolicy switched to use CEL for this part of the validation if you want to follow the same pattern for DNSPolicy

https://github.com/Kuadrant/kuadrant-operator/blob/c831a7005b4bc5e772f6d0f790fb4e2796d79973/api/v1beta2/authpolicy_types.go#L134-L135

Reply:
Created issue to track this https://github.com/Kuadrant/kuadrant-operator/issues/428